src/analytics/index.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  sendAnalyticsEvent as sendAnalyticsTraceEvent,
+  Trace as AnalyticsTrace,
+  TraceEvent as AnalyticsEvent,
+} from '@uniswap/analytics'
+import { atomWithStorage, useAtomValue } from 'jotai/utils'
+import { memo } from 'react'
+
+export { getDeviceId, initializeAnalytics, OriginApplication, user, useTrace } from '@uniswap/analytics'
+
+const allowAnalyticsAtomKey = 'allow_analytics'
+export const allowAnalyticsAtom = atomWithStorage<boolean>(allowAnalyticsAtomKey, true)
+
+export const Trace = memo((props: React.ComponentProps<typeof AnalyticsTrace>) => {
+  const allowAnalytics = useAtomValue(allowAnalyticsAtom)
+  const shouldLogImpression = allowAnalytics ? props.shouldLogImpression : false
+
+  return <AnalyticsTrace {...props} shouldLogImpression={shouldLogImpression} />
+})
+
+Trace.displayName = 'Trace'
+
+export const TraceEvent = memo((props: React.ComponentProps<typeof AnalyticsEvent>) => {
+  const allowAnalytics = useAtomValue(allowAnalyticsAtom)
+  const shouldLogImpression = allowAnalytics ? props.shouldLogImpression : false
+
+  return <AnalyticsEvent {...props} shouldLogImpression={shouldLogImpression} />
+})
+
+TraceEvent.displayName = 'TraceEvent'
+
+export const sendAnalyticsEvent: typeof sendAnalyticsTraceEvent = (event, properties) => {
+  let allowAnalytics = true
+
+  try {
+    const value = localStorage.getItem(allowAnalyticsAtomKey)
+
+    if (typeof value === 'string') {
+      allowAnalytics = JSON.parse(value)
+    }
+    // eslint-disable-next-line no-empty
+  } catch {}
+
```

Comment:
You could just console.error that value was null to remove this eslint ignore

Reply:
The problem is that will log it into sentry and idk if we want this to be logged to sentry because it's not an actual error